app/Http/Controllers/OptionsController.php

```diff
@@ -38,26 +44,19 @@ public function index(PlayerService $player): View
     public function processChangeUsername(Request $request, PlayerService $player): array
     {
         $name = $request->input('new_username_username');
-        $password = $request->input('new_username_password');
         if (!empty($name)) {
-            // Check if password matches.
-            if (!$player->validatePassword($password)) {
-                return array('error' => 'Wrong password!');
-            }
-
             // Check if username validates.
-            if (!$player->validateUsername($name)) {
-                return array('error' => 'Illegal characters in username!');
+            $validationResult = $player->isUsernameValid($name);
+            if (!$validationResult['valid']) {
+                return array('error' => $validationResult['error']);
             }
 
             // Update username
             $player->setUsername($name);
             $player->save();
-
-            return array('success_logout' => 'Settings saved');
         }
 
-        return [];
+        return array('success' => 'Settings saved');
```

Comment:
💡 Request: when you're making the other change(s), could you add __() translation wrappers around this message? 

Note: This wasn't included in the original code so it's not really a comment on this PR, but the end goal is to make all text translatable so when refactoring code I try to add the translation wrappers wherever I can if i see raw strings. Again, not really related to this PR just a quality-of-life thing. Thanks! 😄

Reply:
Indeed, must have forgotten that one as i tried to add them everywhere be default to avoid having to pass again later on these file only to update translations strings.